### PR TITLE
PYIC 4737 Reset appropriate VCs for inherited IDs

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Reset appropriate VCs for inherited IDs

### What changed
Done required changes to reset appropriate VC's for sollowing conditions;

Starting a new P2 proving (with an incomplete/invalid GPG45 profile)
Should clear GPG45 VCs only
V03 mitigation
Should clear GPG45 VCs only
Account intervention
Should clear all VCs
Other identity resets
Should clear all VCs

### Issue tracking

- [PYIC-4737](https://govukverify.atlassian.net/browse/PYIC-4737)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-4737]: https://govukverify.atlassian.net/browse/PYIC-4737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java [hunk 1]
```diff
@@ -111,16 +114,21 @@ void handleRequest_whenNotUserInitiated_shouldDeleteUsersVcsAndReturnNext()
...
         // Act
         JourneyResponse journeyResponse =
                 objectMapper.convertValue(
                         resetIdentityHandler.handleRequest(event, context), JourneyResponse.class);
 
+        try (MockedStatic mock = mockStatic(VcHelper.class)) {
+            // record mock expectations
+            // test code
+            // verify mock
+        }
```
Comment (Joe-Edwards-GDS) [inline lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java]: ??

File: lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java [hunk 2]
```diff
@@ -245,11 +253,75 @@ void handleRequest_whenUserInitiatedF2F_shouldSendEmailWithNullFullNameAndRaiseA
...
+                            .build();
+
+            // Act
+            JourneyResponse journeyResponse =
+                    objectMapper.convertValue(
+                            resetIdentityHandler.handleRequest(event, context),
+                            JourneyResponse.class);
+
+            // Assert
+            dummyStatic.verify(() -> VcHelper.filterVCBasedOnProfileType(any(), any()), times(1));
```
Comment (Joe-Edwards-GDS) [inline lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java:284]: Do we have to mock this? IMO it'd be a better test to assert that we pass the filtered list of `deleteVcStoreItems`, but that might be trickier.

This merely asserts that we call the method - it doesn't actually check that it's used correctly!

If you think it's too difficult, then fine to leave it as-is.

Comment (rpayal) [reply lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java:284]: VcHelper.filterVCBasedOnProfileType already been tested earlier and here we'll only call this when deleteOnlyGPG45Vcs is true.

File: lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java (hunk 1 above)
Comment (rpayal) [reply lambdas/reset-identity/src/test/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandlerTest.java]: Done

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**